### ISSUE TITLE
createConfirmDeviceCredentialIntent is deprecated

### ISSUE THREAD

**Issue by osrl** [auth0/Auth0.Android#653]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/Auth0.Android#readme), [Examples](https://github.com/auth0/Auth0.Android/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/Auth0.Android/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://javadoc.io/doc/com.auth0.android/auth0/latest/index.html) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/Auth0.Android/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

`KeyguardManager`s `createConfirmDeviceCredentialIntent` is deprecated in api level 29.

### Describe the ideal solution

`requireAuthentication` should use `BiometricPrompt` instead of `KeyguardManager`s `createConfirmDeviceCredentialIntent`

**Comment by poovamraj** [comment]
@osrl thanks for raising this. We are aware of this situation and would like to use the BiometricPrompt. We would like to know whether this is currently affecting your workflow? We were not able to produce any issue but would like to be aware if there is any?

We understand this has to be replaced, just want to understand the immediate impact if there is any.

**Comment by osrl** [comment]
There is no immediate impact to me. I will manually use `BiometricPrompt` instead of calling `requireAuthentication`. But I won't be able to use a `BiometricPrompt.CryptoObject` to safely decrypt and encrypt the credentials as recommended. But `CryptoUtil` class might be secure enough. 

`KeyguardManager` isn't necessarily a biometric login. A simple PIN code might be enabled for the lockscreen. So I think it's not secure enough. That's why I'm going to use `BiometricPrompt`. I believe that this issue should be prioritized for this matter.

**Comment by poovamraj** [comment]
@osrl, Sorry on the late response, one of the changes we can look into is to provide an option on how secure the local authentication type has to be (i.e) PIN, fingerprint or other biometrics. This is the option provided in iOS. Are you looking for something similar?

**Comment by osrl** [comment]
yes. I think the same experience should apply to Android as well. But as I said, I've already used BiometricPrompt instead of requireAuthentication myself.

**Comment by poovamraj** [comment]
Hi @osrl we will leave this issue open as we are planning to migrate to BiometricPrompt as well. We would love hear feedback from the community as well.

**Comment by poovamraj** [comment]
Hey @osrl thanks a lot for this feedback. We have internally created a ticket (SDK-4571) to implement this soon. We will close this issue for now. Please feel free to comment here and we can reopen it.
